packages/router/useRouteParams/index.test.ts

```diff
@@ -238,6 +238,27 @@ describe('useRouteParams', () => {
     expect(onUpdate).toHaveBeenCalledTimes(1)
   })
 
+  it('should trigger effects only once with getter object as watch source', async () => {
+    const route = getRoute({ page: '1' })
+    const router = { replace: (r: any) => {
+      Object.keys(r.params).forEach(paramsKey => r.params[paramsKey] = String(r.params[paramsKey]))
+      return Object.assign(route, r)
+    } } as any
```

Comment:
The previous mock router implementation is flawed.
The previous implementation preserves transforms, but vue-router's `.params` and `.query` would always be strings.